src/monitor-plan-workspace/monitor-plan.service.ts

```diff
@@ -512,8 +988,51 @@ export class MonitorPlanWorkspaceService {
     return dto;
   }
 
+  private getYearAndQuarterFromDate(date: Date) {
+    if (!date) return [Infinity, Infinity];
+
+    return [date.getUTCFullYear(), Math.floor(date.getUTCMonth() / 3) + 1];
+  }
+
+  private groupUnitsAndUnitStackConfigsByPeriodAndUnit(
```

Comment:
This is just an initial grouping that will _only_ be used by `mergePartialConfigurations`, which will try to merge the groupings if the locations intersect and the periods intersect. In the example case, item 3 would be grouped with item 2, then items 1 and 4 would later be merged because their stackpipes intersect with the first group.

I'll add some function documentation to make it clear that it's only meant to be used as such.

Reply:
More important than the initial grouping, I think, is transforming the items into the structure expected by `mergePartialConfigurations`. I think we could do away with the grouping, actually, if it's more confusing than helpful.

Reply:
No need to change anything.  I just wanted to make sure a hole did not exist.